### PR TITLE
Fix Default Configuration and Add Note to README

### PR DESCRIPTION
**Pull request by mcdonnnj**
# 

## 🗣 Description
This PR fixes `providers.tf` to correctly use the `aws_region` variable defined in `variables.tf` per a conversation with @dav3r . I also added a note in the README about requiring a `terraform init` in every Terraform code containing directory to successfully pass `pre-commit` per that same conversation. I added defaults and type to variables in `variables.tf` missing them. Lastly I added the `-upgrade=true` switch to the Terraform initialization in the GitHub Actions workflow since we are caching that to ensure the latest version is used.

## 💭 Motivation and Context
@dav3r ran into issues running `pre-commit` on a derived repository and this led to a discussion about changes.

## 🧪 Testing
Automated testing passes.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,12 @@ module "example" {
 | private_ip | The private IP of the EC2 instance |
 | subnet_id | The ID of the subnet where the EC2 instance is deployed |
 
+## Notes ##
+
+Running `pre-commit` requires running `terraform init` in every directory that
+contains Terraform code. In this repository these are the main directory and the
```
Comment (dav3r) [inline README.md]: There should be a comma after `In this repository`.

File: README.md [hunk 2]
```diff
@@ -54,6 +54,12 @@ module "example" {
...
+`examples/default_vpc` directory.
```
Comment (dav3r) [inline README.md]: I'd probably reword this in a way such that it won't necessarily have to be rewritten in downstream repos:
> In this repository, these are the main directory and every directory under `examples/`.

**Review by dav3r — APPROVED** [review]
Thanks for cleaning this up- looks good! 🍔 
Please see the minor change request I mentioned some text in the README.

File: README.md (hunk 1 above)
Comment (mcdonnnj) [reply README.md]: Fixed in a2ec526.

File: README.md (hunk 2 above)
Comment (mcdonnnj) [reply README.md]: Fixed in a2ec526.

File: README.md (hunk 1 above)
Comment (dav3r) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (dav3r) [reply README.md]: 👍 Thanks!

**Review by jsf9k — APPROVED** [review]
:+1:

**Pull request merged**